### PR TITLE
First steps

### PR DESCRIPTION
**Pull request by denis-onder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,49 @@
 # NachoNight Authentication API
+
+> Please remember to run `npm install` to install all the necessary dependencies.
```
Comment (gmemstr) [inline README.md]: Don't need to blockquote this stuff :)

File: README.md [hunk 2]
```diff
@@ -1 +1,49 @@
...
+DB_PASSWORD=
+DB_HOST=
+DB_PORT=
+```
+
+---
+
+### Todos:
+
+- Run a Postgres Docker instance
```
Comment (gmemstr) [inline README.md]: Should be checkboxes like so:

` - [ ] Item`

File: app/middleware/logger.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  The function also takes in the app as an argument,
+  which will be passed in from the function in the
+  index.js file contained in the same folder
+*/
+const morgan = require("morgan");
+const fs = require("fs");
+const path = require("path");
+
+module.exports = app => {
+  const folderPath = path.resolve(__dirname, "../logs");
```
Comment (gmemstr) [inline app/middleware/logger.js]: Can we push the configuration to another directory to make gathering / backing up the logs easier?

File: app/validation/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      this.errors.emailInvalid = "Please provide a valid email address.";
+    // Password validation
+    if (!isValid(data.password))
+      this.errors.passwordInputInvalid =
+        "The provided input for the password is not valid.";
+    if (isEmpty(data.password))
+      this.errors.passwordEmpty = "Please provide your password.";
+    if (!isLength(data.password, { min: 8, max: 32 }))
+      this.errors.passwordLength =
+        "Please provide password between 8 and 32 characters long.";
```
Comment (gmemstr) [inline app/validation/index.js]: Should be no limit on how long a password can be

File: .env.example [hunk 5]
```diff
@@ -0,0 +1,6 @@
+DB_NAME=
+DB_USER=
+DB_PASSWORD=
+DB_HOST=
+DB_PORT=
+SECRET_OR_KEY=
```
Comment (gmemstr) [inline .env.example]: Is it the secret, or the key? Let's pick one ;)

File: README.md [hunk 6]
```diff
@@ -1 +1,49 @@
 # NachoNight Authentication API
+
+> Please remember to run `npm install` to install all the necessary dependencies.
+
+---
+
+### Requirements:
```
Comment (gmemstr) [inline README.md]: Versions of these requirements?

Comment (gmemstr) [reply README.md]: Also, which ones are required for production?

File: README.md [hunk 7]
```diff
@@ -1 +1,49 @@
...
+
+---
+
+### Todos:
+
+- Run a Postgres Docker instance
+
+- Write a docker-compose file to run the app dockerized
+
+---
```
Comment (gmemstr) [inline README.md]: Trailing

File: app/db/models/user.model.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  verified: {
+    type: Sequelize.BOOLEAN,
+    defaultValue: false
+  },
+  banned: {
+    type: Sequelize.BOOLEAN,
+    defaultValue: false
+  },
+  createdAt: {
```
Comment (gmemstr) [inline app/db/models/user.model.js]: Just `created` is fine here :)

File: app/controller.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        email: req.body.email,
+        password: hashSync(req.body.password, 10)
+      });
+      console.log(`${newUser.email} has registered.`);
+      return res.status(200).json(newUser);
+    } catch (error) {
+      return res.status(500).json(error);
+    }
+  }
+  async login(req, res) {
```
Comment (gmemstr) [inline app/controller.js]: Should add a TODO to handle banned accounts here

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,13 @@
+FROM node:10.16.3-alpine
```
Comment (gmemstr) [inline Dockerfile]: We should really author a Docker compose file to simplify setting up the database as well

File: app/db/index.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const Sequelize = require("sequelize");
+const { name, username, password, host } = require("../config").database;
+
+module.exports = new Sequelize(name, username, password, {
+  host,
+  dialect: "postgres"
```
Comment (gmemstr) [inline app/db/index.js]: Would it be hard to make this configurable? Then we could be relatively database agnostic just in case.

File: app/middleware/logger.js (hunk 3 above)
Comment (denis-onder) [reply app/middleware/logger.js]: Where would you like them?

File: Dockerfile (hunk 10 above)
Comment (denis-onder) [reply Dockerfile]: I agree, to be honest I only wrote the dockerfile for testing it out on my own machine

File: app/middleware/logger.js (hunk 3 above)
Comment (gmemstr) [reply app/middleware/logger.js]: Ideally we would be able to set the directory in the configuration, but you can ignore this for now - could just set a symlink on the server.

Comment (denis-onder) [reply app/middleware/logger.js]: Will leave it alone for the time being then :D

**Comment by denis-onder** [comment]
Issue #<issue number>

TL;DR - What does this implement?
---------------------------------

How to Test
-----------

Worries / Concerns
------------------

Checklist
---------

 - [ ] Do we need to test this?
 - [ ] Were tests written?
 - [ ] Does this meet coding standards?
 - [ ] Is this is code change?
 - [ ] Is this a meta change (e.g issue template)?
 - [ ] Is this pull request ready for review?
 - [ ] Is this a breaking change (e.g not backwards compatible with clients)?
 
ESLint Output
-------------

```
// Paste here
```

Reviewers
---------

(Tag people here)
@gmemstr
